packages/language-server/src/plugins/PluginHost.ts

```diff
@@ -105,12 +104,25 @@ export class PluginHost {
 			});
 		}
 
-		let flattenedCompletions = flatten(completions.map((completion) => completion.result.items));
+		let flattenedCompletions = completions.flatMap((completion) => completion.result.items);
 		const isIncomplete = completions.reduce(
 			(incomplete, completion) => incomplete || completion.result.isIncomplete,
 			false as boolean
 		);
 
+		// If the result is incomplete, we need to filter the results ourselves
+		// to throw out non-matching results. VSCode does filter client-side,
+		// but other IDEs might not.
+		if (isIncomplete && this.pluginHostConfig.filterIncompleteCompletions) {
+			const offset = document.offsetAt(position);
+			// Assumption for performance reasons:
+			// Noone types import names longer than 20 characters and still expects perfect autocompletion.
+			const text = document.getText().substring(Math.max(0, offset - 20), offset);
+			const start = regexLastIndexOf(text, /[\W\s]/g) + 1;
+			const filterValue = text.substring(start).toLowerCase();
+			flattenedCompletions = flattenedCompletions.filter((comp) => comp.label.toLowerCase().includes(filterValue));
+		}
+
```

Comment:
Unrelated to the Lodash change, this part of the code seems to have been removed by accident at some point and the test suite for the pluginhost brought it up